src/main/java/com/mongodb/kafka/connect/source/StartedMongoSourceTask.java

```diff
@@ -210,68 +211,80 @@ private List<SourceRecord> pollInternal() {
         createValueSchemaAndValueProvider(sourceConfig);
 
     List<SourceRecord> sourceRecords = new ArrayList<>();
-    getNextBatch()
-        .forEach(
-            changeStreamDocument -> {
-              Map<String, String> sourceOffset = new HashMap<>();
-              sourceOffset.put(ID_FIELD, changeStreamDocument.getDocument(ID_FIELD).toJson());
-              if (isCopying) {
-                sourceOffset.put(COPY_KEY, "true");
-              }
+    Iterator<BsonDocument> batchIterator = getNextBatch().iterator();
+    BsonDocument changeStreamDocument;
```

Comment:
[nit] Do we need to define this outside the while loop?

Reply:
Err nope 😆 I'll update